app/services/course/assessment/submission/zip_download_service.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::ZipDownloadService
+  class << self
+    # Downloads the submissions and zip them.
+    #
+    # @param [CourseUser] course_user The course user downloading the submissions.
+    # @param [Course::Assessment] assessment The assessments to download submissions from.
+    # @param [String|nil] students The subset of students whose submissions to download.
+    # Accepted values: 'my_students', 'students', 'others'
+    # @return [String] The path to the zip file.
+    def download_and_zip(course_user, assessment, students)
+      service = new(course_user, assessment, students)
+      ActsAsTenant.without_tenant do
+        service.send(:download_to_base_dir)
+      end
+      service.send(:zip_base_dir)
+    end
+  end
+
+  STUDENTS = { my: 'my', phantom: 'phantom' }.freeze
+
+  private
+
+  def initialize(course_user, assessment, students)
+    @course_user = course_user
+    @assessment = assessment
+    @students = students
+    @base_dir = Dir.mktmpdir('coursemology-download-')
+  end
+
+  # Downloads each submission to its own folder in the base directory.
+  def download_to_base_dir
+    submissions = @assessment.submissions.by_users(student_ids)
```

Comment:
Just a random note here, the upgrade to Rails 4.2.8 does affect the construction of some of these queries. Not sure if it's possible, but if you do, try to guard this with a spec. One way to do it is to combine this line and `student_ids` into the same method and test whether that method loads the correct set of submissions.